### PR TITLE
login: make RuntimeDirectoryInodesMax= support K, G, M suffixes

### LINKED ISSUE #23017: can RuntimeDirectoryInodesMax=400k is correctly parsed
**systemd version the issue has been seen with**
 > systemd main branch

in [src/login/logind-gperf.gperf](https://github.com/systemd/systemd/blob/a520bb665417af7aa98dcb983f3583659bbab807/src/login/logind-gperf.gperf) 
`Login.RuntimeDirectoryInodesMax,    config_parse_uint64,                0, offsetof(Manager, runtime_dir_inodes)`
use config_parse_uint64 to parse argv, can RuntimeDirectoryInodesMax=400k is correctly parsed.

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #23017.

### CODE REVIEW

File: src/login/logind.conf.in [hunk 1]
```diff
@@ -42,7 +42,7 @@
 #IdleAction=ignore
 #IdleActionSec=30min
 #RuntimeDirectorySize=10%
-#RuntimeDirectoryInodesMax=400k
+#RuntimeDirectoryInodesMax=409600
```
Comment (yuwata) [inline src/login/logind.conf.in]: The header of this file says the value is compile-time default. But `RuntimeDirectoryInodesMax=` is calculated from `RuntimeDirectorySize=` by default... So,
1. keep this as is,
2. make this blank,
3. add a brief comment,
4. update the header.

Which option is favorable?

**Comment by keszybz** [comment]
Can we just change the parser to accept suffixes? We have this fancy parser, so let's use it. If we *documented* this, it means that it's at least plausibly useful to have the rich syntax.

**Comment by yuwata** [comment]
OK. That sounds good for me.

**Review by anitazha — APPROVED** [review]

File: src/login/logind.conf.in (hunk 1 above)
Comment (anitazha) [reply src/login/logind.conf.in]: I like the comment you added. We have something similar with `#DefaultLimitMEMLOCK=8M` where it has a value written in the conf but the actual default value is 1/8 of memory.

File: src/login/logind.conf.in [hunk 2]
```diff
@@ -42,6 +42,8 @@
 #IdleAction=ignore
 #IdleActionSec=30min
 #RuntimeDirectorySize=10%
+# Unlike the other settings, 400k for RuntimeDirectoryInodesMax= below is not
+# the default value. Defaults to RuntimeDirectorySize= divided by 4096.
 #RuntimeDirectoryInodesMax=400k
```
Comment (poettering) [inline src/login/logind.conf.in]: i think we should not mention any "400k" here if it's not actually used anywhre. just leave the rhs of the assignment empty

Comment (yuwata) [reply src/login/logind.conf.in]: Done.

**Comment by poettering** [comment]
hence: lgtm

**Pull request merged**